peazip-sources/unit_pea.pas

```diff
@@ -7268,13 +7273,16 @@ procedure readconf_relativeline(nlines:integer; var dummy:ansistring);
    {$ENDIF}
    {$IFDEF NETBSD}
    if (confpath='appdata') or (confpath='"appdata"') or (confpath='''appdata''') or (confpath='%appdata%') then confpath:=GetEnvironmentVariable('HOME')+'/.PeaZip/';
+   {$ENDIF}  
+   {$IFDEF DARWIN}
+   if (confpath='appdata') or (confpath='"appdata"') or (confpath='''appdata''') or (confpath='%appdata%') then confpath:=GetEnvironmentVariable('HOME')+'/.PeaZip/';
    {$ENDIF}
    if not(directoryexists(confpath)) then mkdir(confpath);
    if (confpath[1]='.') and (confpath[2]='.') then confpath:='..'+directoryseparator+confpath; //relative path, needs to be adjusted since pea is in a subfolder of peazip path
    confpath:=expandfilename(confpath);
    if confpath[length(confpath)]<>directoryseparator then confpath:=confpath+directoryseparator;
-   if not(directoryexists(confpath)) then confpath:=executable_path; //if alternative configuration directory does not exist or is not accessible, use res path
-   persistent_source:=confpath+'rnd';
+   if not(directoryexists(confpath)) then confpath:=resource_path; //if alternative configuration directory does not exist or is not accessible, use res path
+   persistent_source:=resource_path+'rnd';

```

Comment:
persistent_source: should be in confpath which is unique for each user. If it is necessary, I'll add that modification in a conditional compilation section spèecific to Darwin
   {$IFDEF DARWIN}
   persistent_source:=resource_path+'rnd';
   {$ENDIF}